### ISSUE TITLE
Non changing transformation

### ISSUE THREAD

**Issue by cstamas** [l2x6/pom-tuner#162]
For example https://github.com/l2x6/pom-tuner/issues/160 should be "non changing transformation".

Check:
1. get effective model
2. apply transformation
3. get effective model, should be same as in 1.

**Comment by ppalaga** [comment]
Could you plz explain the point 4 in more detail? What do you mean with 

> disregard "management" dep/plg, so consider consumer POM only?

For what purposes "disregard"? 
What is "consumer POM"?

**Comment by cstamas** [comment]
@ppalaga sorry, all these issues were created hastily by me. You are right, this is a "low level" lib. I will pull something a top of it in toolbox.

**Comment by cstamas** [comment]
In this case, my point was: this lib should work "hand in hand" with Maven, and could guarantee that a transformation is "safe" if:
* before change ask for "effective model"
* apply change
* after change again ask for "effective model"
* perform comparisons, the "real values" (so those like plugins and deps, but not management) should be same. If this is true, the transformation is "safe" (as nothing changes re end result, would be like "refactoring" of POM).

**Comment by ppalaga** [comment]
Thanks for the explanation, now I see what you mean. So to make what you ask for, pom-tuner should offer means to get the effective model of a pom (equivalent or the same as `mvn help:effective-pom`), right? Everything else seems to be available.

**Comment by ppalaga** [comment]
I do not see much hope to see this as a pursuable now or in the future. `pom-tuner` transformers operate on DOM only. There is no support planned for accessing effective POM model.

**Comment by cstamas** [comment]
Yup, agreed. This is more for some "maven specific bit". Btw, domtrip is similarly organized: there is domtrip "core" that is really about DOM-like stuff, and there is domtrip-maven that adds support.

And third thing I wanted to share: for eff model you could use MIMA mmr extension:
https://github.com/maveniverse/mima/tree/main/extensions/mmr

**Comment by ppalaga** [comment]
Yeah, thanks, I am already using mmr in one of my pet projects.
